ext/opcache/Optimizer/zend_inference.c

```diff
@@ -3924,6 +3927,300 @@ void zend_inference_check_recursive_dependencies(zend_op_array *op_array)
 	free_alloca(worklist, use_heap);
 }
 
+int zend_may_throw(const zend_op *opline, zend_op_array *op_array, zend_ssa *ssa)
+{
+	uint32_t t1 = OP1_INFO();
+	uint32_t t2 = OP2_INFO();
+
+    if (opline->op1_type == IS_CV) {
+		if (t1 & MAY_BE_UNDEF) {
+			switch (opline->opcode) {
+				case ZEND_UNSET_VAR:
+				case ZEND_ISSET_ISEMPTY_VAR:
+					if (opline->extended_value & ZEND_QUICK_SET) {
+						break;
+					}
```

Comment:
Off topic: I think we might want to split off `ZEND_UNSET_CV` and `ZEND_ISSET_ISEMPTY_CV` opcodes. The code-path of QUICK_SET is completely independent, so I don't think it makes sense to have this in the same opcode as varvar isset/unset.